src/sentry/monitors/logic/mark_failed.py

```diff
@@ -114,35 +112,35 @@ def mark_failed_threshold(
         monitor_env.environment.name,
         str(monitor_env.last_state_change),
     ]
+
     for previous_checkin in previous_checkins:
-        reason = get_reason_from_checkin(previous_checkin)
-        occurrence_context = get_occurrence_context_from_checkin(previous_checkin)
-        create_issue_platform_occurrence(monitor_env, reason, occurrence_context, fingerprint)
+        create_issue_platform_occurrence(previous_checkin, fingerprint)
 
     monitor_environment_failed.send(monitor_environment=monitor_env, sender=type(monitor_env))
 
     return True
 
 
 def mark_failed_no_threshold(
-    monitor_env: MonitorEnvironment,
-    last_checkin=None,
-    reason=MonitorFailure.UNKNOWN,
-    occurrence_context=None,
+    failed_checkin: MonitorCheckIn,
+    ts: datetime | None,
 ):
     from sentry.signals import monitor_environment_failed
 
-    if last_checkin is None:
+    monitor_env = failed_checkin.monitor_environment
+
+    if ts is None:
         next_checkin_base = timezone.now()
         last_checkin = monitor_env.last_checkin or timezone.now()
     else:
-        next_checkin_base = last_checkin
+        next_checkin_base = ts
+        last_checkin = ts
 
-    new_status = MonitorStatus.ERROR
-    if reason == MonitorFailure.MISSED_CHECKIN:
-        new_status = MonitorStatus.MISSED_CHECKIN
-    elif reason == MonitorFailure.DURATION:
-        new_status = MonitorStatus.TIMEOUT
+    failed_status_map = {
+        CheckInStatus.MISSED: MonitorStatus.MISSED_CHECKIN,
+        CheckInStatus.TIMEOUT: MonitorStatus.TIMEOUT,
+    }
```

Comment:
put this up top as a constant?

Reply:
sure

Reply:
Actually I think i'm just gonna leave it here, we don't need it in any of the other functions and I would rather it not be "exported"